Initial working commit

milljm opened the pull request:
To use/test on Sawtooth:

```bash
module load use.moose
module load moosetainer
```

Create a source file which will function to load the above modules when QSUB launches on nodes:

```pre
$ cat my_apptainer_env.sh
source /etc/profile
export MOOSE_COMMAND=${MOOSE_COMMAND:-"null"}
module load use.moose openmpi
module load moosetainer
```

Then be sure to tell TestHarness to source this file while using pbs options:

```bash
./run_tests --pbs alltests --pbs-pre-source /abs/path/to/my_apptainer_env.sh
```

Closes #25038

Commit:
Initial working commit

To use/test on Sawtooth:

```bash
module load use.moose
module load moosetainer
```

Create a source file which will function to load the above modules
when QSUB launches on nodes:

```pre
$ cat my_apptainer_env.sh
source /etc/profile
export MOOSE_COMMAND=${MOOSE_COMMAND:-"null"}
module load use.moose openmpi
module load moosetainer
```

Then be sure to tell TestHarness to source this file while using pbs
options:

```bash
./run_tests --pbs alltests --pbs-pre-source /abs/path/to/my_apptainer_env.sh
```

Closes #25038

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1683080/) on c02d487 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25133/coverage/xfem)

</details>



This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1683062/) on c02d487 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25133/site)

This comment will be updated on new commits.

Commit:
Add PBS wrapper scripts

Allow for the execution of PBS tools from within the container.
Mount points are still necessary from the host machine (e.g:
`-B /var/spool/pbs/aux`)

Commit:
Check for PBS_HOST

Commit:
Syntax changes

Reword a few things for easier understanding

Commit:
skip PBS unittest if PBS_HOST is not available

Now that we are including `qstat, qsub` within the container, we need
to skip PBS unittests for other reasons why PBS may not be available.

moosebuild commented:
All [jobs](https://civet.inl.gov/event/141886/) on dd4f700 : canceled by @milljm

Commit:
Work with relocatble binaries

QueueManager to understand when to run `run_tests` vs `binary-opt`.

moosebuild commented:
Job [WIP Sawtooth](https://civet.inl.gov/job/1752964/) on dc7811b : canceled by @milljm
